tools/adventure-pack/goodies/typescript/traverseLevelOrder/index.ts

```diff
@@ -0,0 +1,23 @@
+export function* traverseLevelOrder<
+  T extends { left?: T | null | undefined; right?: T | null | undefined },
+>(root: T | null | undefined): Generator<T[], void, void> {
+  if (!root) {
+    yield [];
+  }
+
+  const queue = [root];
+  while (queue.length > 0) {
+    const level = [];
+    const size = queue.length;
+    for (let i = 0; i < size; ++i) {
+      const node = queue.shift();
+      if (node) {
+        level.push(node);
+        queue.push(node.left, node.right);
+      }
+    }
+    if (level.length > 0) {
+      yield level;
+    }
+  }
+}
```

Comment:
We have snapshots now, that verify that the goody code can be correctly extracted and rendered. You will need to run `yarn package-goodies:test -u` to update the snapshot tests so the health report becomes green!

Reply:
thank you! I was wondering where I needed to run this test 😄

Reply:
@miorel I am getting this error for the snapshots

![image](https://github.com/user-attachments/assets/622ad4d2-5b1a-4d45-a282-709b2719edf1)

Reply:
Do you know if your system is using Windows line endings or something like that?

Reply:
I couldn't run `yarn package-goodies:test -u` and had to run something like `NODE_OPTIONS=--experimental-vm-modules yarn jest --color --testPathIgnorePatterns="/goodies/" -u`

Reply:
What error do you get when you run `yarn package-goodies:test -u`?